OpenEdX/Router.swift

```diff
@@ -380,6 +380,10 @@ public class Router: AuthorizationRouter,
             lastVisitedBlockID: lastVisitedBlockID
         )
         navigationController.pushViewController(controller, animated: true)
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            Container.shared.resolve(PushNotificationsManager.self)?.performRegistration()
```

Comment:
The push notification registration is happening from the course home screen, is that intended behavior?

Reply:
Yes, it's a UX improvement.
Users automatically reject a push notification request if it appears when the app is launched.